addons/account/models/account_account.py

```diff
@@ -506,12 +506,18 @@ def default_get(self, default_fields):
         return super(AccountAccount, contextual_self).default_get(default_fields)
 
     @api.model
-    def _order_by_frequency_per_partner(self, company_id, partner_id, move_type=None, limit=None):
+    def _order_by_frequency_per_partner(self, company_id, partner_id, move_type=None, only_one=False):
         """
         Returns the accounts ordered from most used to least used for a given partner
-        and filtered according to the move type.
+        and filtered according to the move type. If only_one is True, returns only
+        the most used account.
         """
-        limit = "LIMIT 1" if limit is not None else ""
+        if only_one:
+            limit = "LIMIT 1"
+            join = "INNER JOIN"
+        else:
+            limit = ""
+            join = "LEFT JOIN"
```

Comment:
What? Always do a (~~inner~~ standard) join, no matter what the limit is

And instead of changing the `limit` param, what about you use it correctly?

```python
limit = f"LIMIT {limit:d}" if limit is not None else ""
```